layerfetcher/layers.go

```diff
@@ -403,74 +264,80 @@ func (l *Logic) PollLayerBlocks(ctx context.Context, layerID types.LayerID, hash
 
 	// send a request to the first peer of the list to get blocks data.
 	// todo: think if we should aggregate or ask from multiple peers to have some redundancy in requests
-	numHashes := len(hashes)
-	for hash, remotePeers := range hashes {
-		peer := remotePeers[0]
-		if hash == mesh.EmptyLayerHash {
-			l.receiveBlockHashes(ctx, layerID, peer, numHashes, nil, ErrZeroLayer)
-			continue
-		}
+	for _, p := range remotePeers {
```

Comment:
previous implementation used hashes to avoid requesting all peers for the same data, now as I understand the new flow if all peers will return same blocks than they will be queried again for all the blocks
right?

Reply:
indeed. this is the price of this PR. 
previous behavior
1. RTT to N peers to get layer hash, payload: N*32bytes (Hash32)
2. RTT to M peers to grab layer block IDs, payload: 32byte*NumBlocks*M, M is probably 1 or 2
3. RTT to X random peers grab block contents
behavior after this PR
1. RTT to N peers to get layer block IDs, payload: 32bytes*NumBlocks*N
2. RTT to X random peers grab block contents

but because we do have a upper-bound for N (num peers), i think this is acceptable. 

another option is to use half of the peers as a future optimization? for now i feel that simplicity is most important.